ants/ant.cpp

```diff
@@ -66,59 +78,33 @@ ant step(ant curr, grid landscape, coord spawn_point, int gen_flag){
     left.x = curr.pos.x - 1;    left.y = curr.pos.y;
     right.x = curr.pos.x + 1;   right.y = curr.pos.y;
 
+    coord last = curr.antpath.back();
     // determine possible movement
-    for (int i = 0; i < 4; i++){
-        switch(i){
-        // up case
-        case 0:
-            if (up.x == curr.antpath.back().x &&
-                up.y == curr.antpath.back().y){
-            }
-            else{
-                next_step[pcount] = up;
-                pcount++;
-            }
-            break;
-
-        // down case
-        case 1:
-            if (down.x == curr.antpath.back().x &&
-                down.y == curr.antpath.back().y){
-            }
-            else{
-                next_step[pcount] = down;
-                pcount++;
-            }
-            break;
-
-        // left case
-        case 2:
-            if (left.x == curr.antpath.back().x &&
-                left.y == curr.antpath.back().y){
-            }
-            else{
-                next_step[pcount] = left;
-                pcount++;
-            }
-            break;
-
-        // right case
-        case 3:
-            if (right.x == curr.antpath.back().x &&
-                right.y == curr.antpath.back().y){
-            }
-            else{
-                next_step[pcount] = right;
-                pcount++;
-            }
-            break;
-
-        }
+    // up case
+    if (last != up) {
+        next_step[pcount] = up;
+        pcount++;
+    }
+
+    // down case
+    if (last != down) {
+        next_step[pcount] = down;
+        pcount++;
+    }
+
+    if (last != left) {
+        next_step[pcount] = left;
+        pcount++;
+    }
 
+    // right case
+    if (last != right) {
+        next_step[pcount] = right;
+        pcount++;
     }
 
     static std::random_device rd;
-    int seed = rd();
+    auto seed = rd();
```

Comment:
It's better to use auto when storing a result whose type you're not sure about. In this case it wasn't a big deal (rd() return `unsigned int`), but you could've truncated the initial seed, if it was larger (but implicitly convertible) to `int`.